### PR TITLE
man: add simple build test for example code

### LINKED ISSUE #32054: CI build coverage for examples embedded in documentation is missing
### systemd version the issue has been seen with

git main

### Used distribution

_No response_

### Linux kernel version used

_No response_

### CPU architectures issue was seen on

None

### Component

_No response_

### Expected behaviour you didn't see

Just to keep track of @evverx's findings here:

https://github.com/systemd/systemd/pull/32030#issuecomment-2032540493

This really should be addressed.

### Unexpected behaviour you saw

see above

### Steps to reproduce the problem

see above

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
foo
```

### PR DESCRIPTION
**Pull request by yuwata**
closes #32054.

### CODE REVIEW

**Comment by yuwata** [comment]
@mrc0mmand @evverx This is now ready for review. Please take a look.

File: man/sd_bus_service_reconnect.c [hunk 1]
```diff
@@ -90,13 +87,20 @@ static const sd_bus_vtable vtable[] = {
 static int setup(object *o);
 
 static int on_disconnect(sd_bus_message *message, void *userdata, sd_bus_error *ret_error) {
-  check(setup((object *)userdata));
+  int r;
+
+  r = setup((object *)userdata);
+  if (r < 0)
+    log_error(r, "setup()");
```
Comment (mrc0mmand) [inline man/sd_bus_service_reconnect.c]: Missing `return`? But I guess it doesn't matter here.

File: man/sd_bus_service_reconnect.c [hunk 2]
```diff
@@ -107,19 +111,25 @@ static int request_name_callback(sd_bus_message *m, void *userdata, sd_bus_error
 
   printf("Failed to request name: %s\n", error->message);
   object *o = userdata;
-  check(sd_event_exit(*o->event, -sd_bus_error_get_errno(error)));
+  r = sd_event_exit(*o->event, -sd_bus_error_get_errno(error));
+  if (r < 0)
+    log_error(r, "sd_event_exit()");
```
Comment (mrc0mmand) [inline man/sd_bus_service_reconnect.c]: ditto

File: man/meson.build [hunk 3]
```diff
@@ -248,3 +248,89 @@ update_man_rules = custom_target(
...
+                        ],
+                },
+        ]
+endif
+
+default_args = [
+        cc.cmd_array(),
+        '-c',
+        '-x', 'c',
+        '-std=c11',
```
Comment (poettering) [inline man/meson.build]: this should be dropped, no? you override it below anyway, after all

**Comment by yuwata** [comment]
@poettering @mrc0mmand Thank you for the review. Addressed. PTAL.

**Review by mrc0mmand — APPROVED** [review]
LGTM, thank you!

**Comment by evverx** [comment]
Regarding `-D_GNU_SOURCE` I don't think it's documented anywhere that it should be set explicitly. It seems `#define _GNU_SOURCE 1` or something like that would make examples self-contained.

**Pull request merged**